### PR TITLE
[1.10 Only] Update hub workload pool suffix

### PR DESCRIPTION
**Pull request by JimmyCYJ**

### CODE REVIEW

**Comment by JimmyCYJ** [comment]
DO NOT SUBMIT

File: scripts/asm-installer/asm_vm [hunk 1]
```diff
@@ -946,7 +946,15 @@ uses_environ_workload_identity() {
 
   if [[ "$(retry 2 kubectl get configmap "${CONFIGMAP}" -n istio-system \
     -ojsonpath='{.data.mesh}' | grep -x -c "^trustDomain: ${PROJECT_ID}.hub.id.goog$" \
```
Comment (jasonwzm) [inline scripts/asm-installer/asm_vm:948]: We can achieve the check in one line with an OR in the regex. e.g. https://github.com/GoogleCloudPlatform/anthos-service-mesh-packages/blob/master/scripts/asm-installer/install_asm#L2201

**Review by jasonwzm — CHANGES_REQUESTED** [review]
@jtrbs Please also review.

File: scripts/asm-installer/asm_vm (hunk 1 above)
Comment (JimmyCYJ) [reply scripts/asm-installer/asm_vm]: Done. Please take a look. Thanks!

File: scripts/asm-installer/asm_vm [hunk 2]
```diff
@@ -945,8 +945,11 @@ uses_environ_workload_identity() {
   fi
 
   if [[ "$(retry 2 kubectl get configmap "${CONFIGMAP}" -n istio-system \
-    -ojsonpath='{.data.mesh}' | grep -x -c "^trustDomain: ${PROJECT_ID}.hub.id.goog$" \
-    || true)" -eq 0 ]]; then
+    -ojsonpath='{.data.mesh}' | grep -x -c "^trustDomain: ${PROJECT_ID}.\(hub\|svc\).id.goog$" \
+    || true)" -eq 1 ]]; then
```
Comment (jasonwzm) [inline scripts/asm-installer/asm_vm:949]: I guess we can do `-eq 0` here and we can just `false` below, so we don't need the else below.

**Review by jasonwzm — CHANGES_REQUESTED** [review]

File: scripts/asm-installer/asm_vm (hunk 1 above)
Comment (jasonwzm) [reply scripts/asm-installer/asm_vm]: LGTM

File: scripts/asm-installer/asm_vm (hunk 2 above)
Comment (JimmyCYJ) [reply scripts/asm-installer/asm_vm]: Forgot to change it back. Done.

**Review by jasonwzm — APPROVED** [review]

**Review by jtrbs — APPROVED** [review]
LGTM

**Pull request merged**